sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -358,6 +359,18 @@ abstract class JdbcDialect extends Serializable with Logging{
     new AnalysisException(message, cause = Some(e))
   }
 
+  /**
+   * Convert key-value property pair to string
+   */
+  def convertPropertyPairToString(key: String, value: String): String = {
```

Comment:
Note that, every method in this class is a public API, and we should be very careful when adding new public APIs.

I don't think the two newly added APIs are necessary. They are just used to share the code between different dialects. We should use internal util functions to share code.